pkg/storegateway/bucket_stores.go

```diff
@@ -492,7 +492,7 @@ func (u *BucketStores) getOrCreateStore(userID string) (*store.BucketStore, erro
 		fetcher,
 		u.syncDirForUser(userID),
 		newChunksLimiterFactory(u.limits, userID),
-		store.NewSeriesLimiterFactory(0), // No series limiter.
+		newSeriesLimiterFactory(u.limits, userID),
```

Comment:
Can we make sure that querier will not retry when this limit is enforced?

Reply:
At the moment the querier only retries for missing blocks. 
https://github.com/cortexproject/cortex/blob/master/pkg/querier/blocks_store_queryable.go#L515